### ISSUE TITLE
Question about the Contextual Relation Module in the SAR-net paper.

### ISSUE THREAD

**Issue by hmtrii** [Deepwise-AILab/ChestX-Det-Dataset#1]
Hi, I'm having a problem when trying to implement your proposed network in _A Structure-Aware Relation Network for
Thoracic Diseases Detection and Segmentation_ paper. 
In session 3.2 Contextual Relation Module of the paper, I don't understand clearly **how you create the grid-shaped feature**. _Could you explain to me the operation you did to obtain the grid-shaped feature?_ 
![image](https://user-images.githubusercontent.com/42197959/126041982-1b37d41c-73bd-47d3-a6a0-f6062fef15af.png)
Thank you a lot. ❤️

**Comment by jeline0110** [comment]
Obtaining the grid-shaped feature can be equivalent to obtaining the new grid-shaped boxes from the original bounding boxes of left/right lung, you can refer to the code.py.

**Comment by jeline0110** [comment]
[code.txt](https://github.com/Deepwise-AILab/ChestX-Det-Dataset/files/7632643/code.txt)

**Comment by jeline0110** [comment]
Obtaining the grid-shaped feature can be equivalent to obtaining the new grid-shaped boxes from the original bounding boxes of left/right lung, you can refer to the code.txt

**Comment by DatDoc** [comment]
Hi @jeline0110 , could you public the training code for baseline experimental results in the paper. Thank you.
![image](https://user-images.githubusercontent.com/62165465/144241186-eb6abc1f-eee9-4f49-ab53-0ead284d4296.png)

**Comment by wlufy** [comment]
Hi @hmtrii  @jeline0110 , sorry to disturb you. I have some confusion about the **Anatomical Structure Relation Module** in the SAR-net paper.
The output of pretrained PSPNet is a vector (shape <anatomical_part_nums, h, w>). Based this vector, I notice that author uses coordinate difference to quantify spatial relation with each anatomical part. I don't **how to get the coordinate of each annatomical part**.  Is it to find the circumscribed rectangle for every organ on the binary mask image? Could you help me solve this confusion?
![image](https://user-images.githubusercontent.com/49024246/166896791-d42b2b91-acb2-45a4-bfb9-12e16e4789f0.png)
Thank you!

**Comment by jeline0110** [comment]
Yes, we find the circumscribed rectangle for every organ on the binary mask image.

**Comment by jeline0110** [comment]
> Hi @hmtrii @jeline0110 , sorry to disturb you. I have some confusion about the **Anatomical Structure Relation Module** in the SAR-net paper. The output of pretrained PSPNet is a vector (shape <anatomical_part_nums, h, w>). Based this vector, I notice that author uses coordinate difference to quantify spatial relation with each anatomical part. I don't **how to get the coordinate of each annatomical part**. Is it to find the circumscribed rectangle for every organ on the binary mask image? Could you help me solve this confusion? ![image](https://user-images.githubusercontent.com/49024246/166896791-d42b2b91-acb2-45a4-bfb9-12e16e4789f0.png) Thank you!


Yes, we find the circumscribed rectangle for every organ on the binary mask image.

**Comment by wlufy** [comment]
> > Hi @hmtrii @jeline0110 , sorry to disturb you. I have some confusion about the **Anatomical Structure Relation Module** in the SAR-net paper. The output of pretrained PSPNet is a vector (shape <anatomical_part_nums, h, w>). Based this vector, I notice that author uses coordinate difference to quantify spatial relation with each anatomical part. I don't **how to get the coordinate of each annatomical part**. Is it to find the circumscribed rectangle for every organ on the binary mask image? Could you help me solve this confusion? ![image](https://user-images.githubusercontent.com/49024246/166896791-d42b2b91-acb2-45a4-bfb9-12e16e4789f0.png) Thank you!
> 
> Yes, we find the circumscribed rectangle for every organ on the binary mask image.

I got it! Thanks!
